iml-agent/src/action_plugins/action_plugin.rs

```diff
@@ -64,14 +64,18 @@ pub fn create_registry() -> HashMap<ActionName, Callback> {
         mk_callback(server::stream_fidlists),
     );
 
-    map.insert(
-        "action_warning_stratagem".into(),
-        mk_callback(action_warning::read_mailbox),
-    );
-    map.insert(
-        "action_purge_stratagem".into(),
-        mk_callback(action_purge::read_mailbox),
-    );
+    if action_warning::is_ok() {
```

Comment:
I think you should remove these gates. 

If one of these plugins is called, It will fail until the .so is present, which is not really an issue.

With these gates in place, We will need to implement a feature to enable plugins via signal (or restart and lose all active sessions).

Reply:
done